### PR TITLE
Add generics constraint test

### PR DESCRIPTION
**Pull request by fadimounir**
@rahku @ramarag, Please review. Porting a generics test to the CoreCLR repo. If all looks good, i'll bulk-port more tests using a tool, in the same format as this test.

### CODE REVIEW

**Comment by ramarag** [comment]
please add licences to all the code files, look at the existing ones for the lines to add

File: tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <AssemblyName>Method_Constraints_Negative</AssemblyName>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <SchemaVersion>2.0</SchemaVersion>
+    <ProjectGuid>{95DFC527-4DC1-495E-97D7-E94EE1F7140D}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
```
Comment (ramarag) [inline tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj]: Is this tag required ?

File: tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    <FileAlignment>512</FileAlignment>
+    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <ReferencePath>$(ProgramFiles)\Common Files\microsoft shared\VSTT\11.0\UITestExtensionPackages</ReferencePath>
```
Comment (ramarag) [inline tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj]: Is this path use anywhere ?

File: tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/app.config [hunk 3]
```diff
@@ -0,0 +1,27 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (ramarag) [inline tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/app.config]: is this file still required, i though all dependencies are pulled in via project.json ?

File: tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj (hunk 1 above)
Comment (fadimounir) [inline tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj]: Copy-paste. Didn't bother to change stuff around (didn't really care)

---
In reply to: [58459822](https://github.com/dotnet/coreclr/pull/4071#discussion_r58459822) [](ancestors = 58459822)

File: tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj (hunk 2 above)
Comment (fadimounir) [inline tests/src/Loader/classloader/generics/Constraints/ConstraintsOnMethod/Negative/Method_Constraints_Negative.csproj]: Copy-paste. Didn't bother to change stuff around (didn't really care)

---
In reply to: [58459855](https://github.com/dotnet/coreclr/pull/4071#discussion_r58459855) [](ancestors = 58459855)

**Comment by ramarag** [comment]
@fadimounir : in future please squash your commits before you merge your pull request
thanks
